lib/srv/reexec.go

```diff
@@ -574,6 +585,18 @@ func buildCommand(c *ExecCommand, localUser *user.User, tty *os.File, pty *os.Fi
 		cmd.SysProcAttr = &syscall.SysProcAttr{
 			Setsid: true,
 		}
+
+		if c.ExtraFilesLen > 0 {
+			cmd.ExtraFiles = make([]*os.File, c.ExtraFilesLen)
+			for i := 0; i < c.ExtraFilesLen; i++ {
+				fd := 6 + i
```

Comment:
Why 6?

Reply:
Added a comment that explains

Reply:
So SFTP needs a finite number of file descriptors, but I added `ExecCommand.ExtraFilesLen` so that the code in `reexec.go` doesn't need to be tightly coupled with the SFTP code. If we need to pass down arbitrary file descriptors to the (grand)child process in the future, we can do that with the new code I added.

Reply:
As for adding constants for FDs we use, I do think that's a good idea

Reply:
Added constants as suggested.